File: leetcode/golang/118.杨辉三角.go
func generate(numRows int) [][]int {
    ret := make([][]int, numRows);
    i := 0;
    for i < numRows{
        tmp := make([]int, i + 1);
        j := 0;
        for j < len(tmp) {
            if j == 0 || j == len(tmp) - 1{
                tmp[j] = 1;
            } else {
                tmp[j] = ret[i - 1][j] + ret[i - 1][j - 1];
            }
            j += 1;
        }
        ret[i] = tmp;
        i += 1;
    }
    return ret
}

File: leetcode/python/118.杨辉三角.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ret = []

        for i in range(numRows):
            tmp = [1 for _ in range(i + 1)]
            for j in range(1, len(tmp) - 1):
                tmp[j] = ret[-1][j] + ret[-1][j - 1]
            ret.append(tmp)
        return ret